cmdstanpy/utils.py

```diff
@@ -177,48 +182,75 @@ def cmdstan_path() -> str:
+    :param major: Major version number
+    :param minor: Minor version number
 
-    :return: True if version at or above, else False
+    :return: True if version at or above major.minor, else False.
     """
-    # pylint:disable=bare-except
     try:
-        path = cmdstan_path()
-        makefile = os.path.join(path, 'makefile')
-        if not os.path.exists(makefile):
-            raise ValueError(
-                'CmdStan installation {}: missing makefile'.format(path)
-            )
-        version = None
-        with open(makefile, 'r') as fd:
-            contents = fd.read()
-            start_idx = contents.find('CMDSTAN_VERSION := ') + len(
-                'CMDSTAN_VERSION := '
-            )
-            end_idx = contents.find('\n', start_idx)
-            version = contents[start_idx:end_idx]
-        if version is None:
-            raise ValueError(
-                'Cannot parse version from makefile: {}'.format(makefile)
-            )
-        splits = version.split('.')
-        if len(splits) < 2:
-            raise ValueError(
-                'Cannot parse version from makefile: {}'.format(makefile)
-            )
-        cur_maj = int(splits[0])
-        cur_min = int(splits[1])
+        makefile = os.path.join(cmdstan_path(), 'makefile')
+    except ValueError:
+        get_logger().info(
+            'No CmdStan installation found, '
+            'cannot assert version is less than %d.%d.',
+            major,
+            minor,
+        )
+        return False
 
-        if cur_maj > maj or (cur_maj == maj and cur_min >= min):
-            return True
-    except:  # noqa
-        pass
+    if not os.path.exists(makefile):
+        get_logger().info(
+            'CmdStan installation %s missing makefile, '
+            'cannot assert version is less than %d.%d.',
+            cmdstan_path(),
+            major,
+            minor,
+        )
+        return False
+
+    with open(makefile, 'r') as fd:
+        contents = fd.read()
+
+    start_idx = contents.find('CMDSTAN_VERSION := ')
+    if start_idx < 0:
+        get_logger().info(
+            'Cannot parse version from makefile: %s,'
+            'cannot assert version is less than %d.%d.',
+            makefile,
+            major,
+            minor,
+        )
+        return False
+
+    start_idx += len('CMDSTAN_VERSION := ')
+    end_idx = contents.find('\n', start_idx)
+
+    version = contents[start_idx:end_idx]
+    if version is None or len(version) < 3 or len(version.split('.')) < 2:
+        get_logger().info(
+            'Cannot parse version from makefile: %s,'
+            'cannot assert version is less than %d.%d.',
+            makefile,
+            major,
+            minor,
+        )
+        return False
+
+    splits = version.split('.')
+    cur_major = int(splits[0])
+    cur_minor = int(splits[1])
+    if cur_major < major or (cur_major == major and cur_minor < minor):
+        return True
```

Comment:
I think it would be useful (and easier to test) if we made this two functions. One called like `get_cmdstan_version` which does the above parsing and returns a tuple, and the other `cmdstan_version_before` which calls the first and basically consists of just these two lines